controllers/addon_controller.go

```diff
@@ -123,20 +123,28 @@ func (r *AddonReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 		return reconcile.Result{}, ignoreNotFound(err)
 	}
+	
+	return r.execAddon(ctx, req, log, instance)
+}
 
+func (r *AddonReconciler) execAddon(ctx context.Context, req reconcile.Request, log logr.Logger, instance *addonmgrv1alpha1.Addon) (reconcile.Result, error) {
+	defer func() {
+		if err := recover(); err != nil {
+			fmt.Println("panic occurred during execAdd :", err)
+		}
+	}()
 	// Process addon instance
 	ret, err := r.processAddon(ctx, req, log, instance)
 
-	// Always update status, cache
+	// Always update cache, status
```

Comment:
This comment was updated previously to represent the order of the code, we update status then cache.